code/game/objects/items/devices/radio/radio.dm

```diff
@@ -354,6 +360,13 @@
 	if(radio_freq || !broadcasting || get_dist(src, speaker) > canhear_range)
 		return
 	var/filtered_mods = list()
+
+	if (message_mods[MODE_SING])
+		filtered_mods[MODE_SING] = message_mods[MODE_SING]
+	if (message_mods[WHISPER_MODE])
+		filtered_mods[WHISPER_MODE] = message_mods[WHISPER_MODE]
+	if (message_mods[SAY_MOD_VERB])
+		filtered_mods[SAY_MOD_VERB] = message_mods[SAY_MOD_VERB]
```

Comment:
There has to be a way to do this saner.
Aside from that, is there even a reason we are filtering the input message mods?
I believe talking directly into a radio forwards them all into `talk_into()` anyway.